app/controllers/DatasetController.scala

```diff
@@ -401,26 +405,32 @@ class DatasetController @Inject()(userService: UserService,
           usableDataSource <- dataSource.toUsable ?~> "dataset.notImported"
           dataLayer <- usableDataSource.dataLayers.find(_.name == dataLayerName) ?~> "dataset.noLayers"
           datastoreClient <- datasetService.clientFor(dataset)(GlobalAccessContext)
-          targetMagBbox: BoundingBox = request.body.boundingBox / request.body.mag
-          _ <- bool2Fox(targetMagBbox.size.sorted == Vec3Int(1, 1024, 1024)) ?~> s"Target-mag bbox must be sized 1024×1024×1 (or transposed), got ${targetMagBbox.size}"
-          data <- datastoreClient.getLayerData(organizationName,
-                                               dataset,
-                                               dataLayer.name,
-                                               request.body.boundingBox,
-                                               request.body.mag,
-                                               request.body.additionalCoordinates) ?~> "segmentAnything.getData.failed"
+          targetMagSelectedBbox: BoundingBox = request.body.surroundingBoundingBox / request.body.mag
```

Comment:
I think the check whether the user is logged in and has access to the dataset is missing here.

Reply:
Both is a bit implicit, I’ll admit…
The logged-in check is taken care of by this being a `sil.SecuredAction`.
The user having access to the dataset is taken care of by the `datasetDAO.findOneByNameAndOrganizationName` (which uses a ReadAccessQuery).

Both of these methods are pretty standard across our backend. Still a good idea to ask the question :)